hooks/useCart.js

```diff
@@ -229,6 +229,26 @@ export default function useCart(initialCart = false, options = {}) {
     }
   };
 
+  const applyCredits = async (value, callback) => {
+    setIsLoading(true);
+    try {
+      const response = await fetch(`/api/lib/v1/cart/credits`, {
+        method: "POST",
+        body: JSON.stringify({ credit: value }),
+      }).then((result) => {
+        if (isFunction(callback)) {
+          callback(result);
+        }
+        return result.json();
+      });
```

Comment:
@rasevp ten callback nikdy nebude vediet spracovat result, lebo result je promise. musi to byt prepisane ze:
`
.then(result => result.json())
.then(result => {
     if (isFunction(callback)) {
       callback(result);
     }
     return result;
});
`

Reply:
dakujem, upravene